integration_tests/src/main/python/spark_session.py

```diff
@@ -67,7 +54,10 @@ def _set_all_confs(conf):
     newconf.update(conf)
     for key, value in newconf.items():
         if _spark.conf.get(key, None) != value:
-            _spark.conf.set(key, value)
+            if isinstance(value, str):
```

Comment:
Maybe only use strings in RapidsConf export to JSON to avoid casts on read?

Reply:
I think this is a bug in some JNI interface in `pysparkSparkConf`, not the JSON problem. It makes sense to me to wrap values here when we call the problematic API.